### ISSUE TITLE
Option to halt progress when rate limited

### ISSUE THREAD

**Issue by FRidh** [DeterminateSystems/magic-nix-cache-action#76]
When rate-limited Nix continues building from source instead, as described also in https://github.com/DeterminateSystems/magic-nix-cache-action?tab=readme-ov-file#usage-notes. Is it feasible to halt when this occurs, and then continue after 60 seconds? In this case building from source takes far more time than having to wait once (or more).

**Comment by dbarrosop** [comment]
Yes, current behaviour is a bit unfortunate, specially as it "hards disable" the store which means you can't attempt to copy results that may not be cached automatically (like the check derivation).

**Comment by DaRacci** [comment]
This sounds like a very solid idea, though id imagine it may be a bit hard to implement due to nix probably wanting to timeout a download thats doing nothing but I'm not sure.
